pkg/roll/run_update.go

```diff
@@ -252,14 +252,14 @@ func (u *update) rollLoop(podID types.PodID, hChecks <-chan map[types.NodeName]h
 					"nextAdd":    nextAdd,
 				}).Infof("Adding %d new nodes and removing %d old nodes", nextAdd, nextRemove)
 				if nextRemove > 0 {
-					err = u.rcs.AddDesiredReplicas(u.OldRC, -nextRemove)
+					err = u.rcs.CASDesiredReplicas(u.OldRC, oldNodes.Desired, oldNodes.Desired-nextRemove)
 					if err != nil {
 						u.logger.WithError(err).Errorln("Could not decrement old replica count")
 						break
```

Comment:
is break here still appropriate? This will cause it to continue the watch, I believe. Perhaps `return false` (acting as if the replication was cancelled) is more appropriate?

Maybe in normal circumstances we should `break`, but if we were pre-empted (can be signalled by making the error type different) we should `return false`?